src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -381,54 +388,71 @@ public void Stake(WalletSecret walletSecret)
-                }
-                catch (MinerException me)
-                {
-                    // Miner exceptions should be ignored. It means that the miner
-                    // possibly mined a block that was not accepted by peers or is even invalid,
-                    // but it should not halted the staking operation.
-                    this.logger.LogDebug("Miner exception occurred in miner loop: {0}", me.ToString());
-                    this.rpcGetStakingInfoModel.Errors = me.Message;
-                }
-                catch (ConsensusErrorException cee)
-                {
-                    // All consensus exceptions should be ignored. It means that the miner
-                    // run into problems while constructing block or verifying it
-                    // but it should not halted the staking operation.
-                    this.logger.LogDebug("Consensus error exception occurred in miner loop: {0}", cee.ToString());
-                    this.rpcGetStakingInfoModel.Errors = cee.Message;
-                }
-                catch
+            this.rpcGetStakingInfoModel.Enabled = true;
+            this.stakeCancellationTokenSource =
+                CancellationTokenSource.CreateLinkedTokenSource(new[] { this.nodeLifetime.ApplicationStopping });
+
+            this.stakingLoop = this.asyncLoopFactory.Run(
+                "PosMining.Stake",
+                async token =>
                 {
-                    this.logger.LogTrace("(-)[UNHANDLED_EXCEPTION]");
-                    throw;
-                }
+                    this.logger.LogTrace("()");
+
+                    try
+                    {
+                        await this.GenerateBlocksAsync(walletSecret).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                            // Application stopping, nothing to do as the loop will be stopped.
+                        }
+                    catch (MinerException me)
+                    {
+                            // Miner exceptions should be ignored. It means that the miner
+                            // possibly mined a block that was not accepted by peers or is even invalid,
+                            // but it should not halted the staking operation.
+                            this.logger.LogDebug("Miner exception occurred in miner loop: {0}", me.ToString());
+                        this.rpcGetStakingInfoModel.Errors = me.Message;
+                    }
+                    catch (ConsensusErrorException cee)
+                    {
+                            // All consensus exceptions should be ignored. It means that the miner
+                            // run into problems while constructing block or verifying it
+                            // but it should not halted the staking operation.
+                            this.logger.LogDebug(
```

Comment:
Please configure VS to use 120 chars in a line, not 80. Everyone have big monitors. 
Imho no need to split it that much. 

However it's purely a personal preference question so if you like it as it is- don't change.